suitest.js

```diff
@@ -0,0 +1,209 @@
+// const executeBrightScriptFactory = require('./lib/chains/executeBrightScriptChain');
+const jsExpressionFactory = require('./lib/chains/javascriptExpressionChain');
+// const brightScriptExpressionFactory = require('./lib/chains/brightScriptExpressionChain');
+const elementFactory = require('./lib/chains/elementChain');
+const pollUrlFactory = require('./lib/chains/pollUrlChain');
+const networkRequestFactory = require('./lib/chains/networkRequestChain');
+const videoFactory = require('./lib/chains/videoChain');
+const playstationVideoFactory = require('./lib/chains/playstationVideoChain');
+
+// Constants
+const {ELEMENT_PROP, VALUE} = require('./lib/constants/element');
+const {PROP_COMPARATOR} = require('./lib/constants/comparator');
+const VIDEO_STATE = require('./lib/constants/videoState');
+const CONTENT_MODE = require('./lib/constants/contentMode');
+const ELEMENT_STATE = require('./lib/constants/elementState');
+const TEXT_ALIGNMENT = require('./lib/constants/textAlignment');
+const IMAGE_LOAD_STATE = require('./lib/constants/imageLoadState');
+const BORDER_STYLE = require('./lib/constants/borderStyle');
+const VISIBILITY_STATE = require('./lib/constants/visibilityState');
+const VRC = require('./lib/constants/vrc');
+const KEY = require('./lib/constants/keys');
+const {NETWORK_PROP, NETWORK_METHOD} = require('./lib/constants/networkRequest');
+const HAD_NO_ERROR = require('./lib/constants/hadNoError');
+
+// Network
+const webSocketsFactory = require('./lib/api/webSockets');
+const unusedExpressionWatchersFactory = require('./lib/utils/unusedExpressionWatchers');
+const ipcClient = require('./lib/testLauncher/ipc/client');
+const ipcServer = require('./lib/testLauncher/ipc/server');
+
+// Contexts
+const Context = require('./lib/utils/Context');
+const AuthContext = require('./lib/utils/AuthContext');
+const {configFactory} = require('./config');
+const {setUpRaven} = require('./lib/utils/sentry/Raven');
+const {warnLauncherAndLibHasDiffVersions} = require('./lib/utils/packageMetadataHelper');
+const {createLogger} = require('./lib/utils/logger');
+
+// Publicly available API goes here
+class Suitest {
+	constructor({exitOnError}) {
+		// instance dependencies
+		this.authContext = new AuthContext();
+		this.appContext = new Context();
+		this.pairedDeviceContext = new Context();
+		this.testContext = new Context();
+		this.configuration = configFactory();
+		this.config = this.configuration.config;
+		this.configuration.configurableFields.map(fieldName => {
+			this[`set${fieldName[0].toUpperCase()}${fieldName.slice(1)}`] = (val) => this.configuration.override({[fieldName]: val});
+		});
+
+		// creating methods based on instance dependencies
+		this.logger = createLogger(this.config, this.pairedDeviceContext);
+		this.unusedExpressionWatchers = unusedExpressionWatchersFactory(this);
+		this.webSockets = webSocketsFactory(this);
+		setUpRaven(this.config, this.authContext);
```

Comment:
We are setting up Raven here, before user has chance to call `setDisallowCrashReports`. Will there be an initial call to Raven right away? If I disallow crash reports, will this actually affect Raven?

Reply:
Will check if Raven does any requests on setUp, but in our set up we have 
```
		shouldSendCallback(data) {
					// send report if enabled in config and only uncaught suitest api errors
					return !config.disallowCrashReports && isSuitestUncaughtError(data.exception[0]);
				},
```
so I think if user will change disallowCrashReports latter it should prevent from sending reports